docs/en/observability/aws-serverless-manage-multiline-messages.asciidoc

```diff
@@ -0,0 +1,241 @@
+
+[[aws-serverless-multiline-config-example-java]]
+=== Java stack traces
+
+Java stack traces consist of multiple lines, with each line after the initial line beginning with whitespace, as in this example:
+
+[source, java]
+----
+Exception in thread "main" java.lang.NullPointerException
+        at com.example.myproject.Book.getTitle(Book.java:16)
+        at com.example.myproject.Author.getBookTitles(Author.java:25)
+        at com.example.myproject.Bootstrap.main(Bootstrap.java:14)
+----
+
+This configuration merges any line that begins with whitespace up to the previous line:
+
+[source, yaml]
+----
+multiline:
+  type: pattern
+  pattern: '^\s'
+  negate: false
+  match: after
+----
+
+This is a slightly more complex Java stack trace example:
+
+[source, java]
+----
+Exception in thread "main" java.lang.IllegalStateException: A book has a null property
+       at com.example.myproject.Author.getBookIds(Author.java:38)
+       at com.example.myproject.Bootstrap.main(Bootstrap.java:14)
+Caused by: java.lang.NullPointerException
+       at com.example.myproject.Book.getId(Book.java:22)
+       at com.example.myproject.Author.getBookIds(Author.java:35)
+       ... 1 more
+----
+
+To consolidate these lines into a single event, use the following multiline configuration:
+
+[source, yaml]
+----
+multiline:
+  type: pattern
+  pattern: '^\s+(at|.{3})\s+\\b|^Caused by:'
+  negate: false
+  match: after
+----
+
+In this example, the pattern matches and merges the following lines:
+-   a line that begins with spaces followed by the word `at` or `...`
+-   a line that begins with the words `Caused by:`
+
+NOTE: In Python’s string literals, `\b` is the backspace character (ASCII value 8). As raw strings are not used, Python would convert the `\b` to a backspace. In order for our regular expression to match as expected, you need to escape the backslash `\` in `\b` to `\\b`, which will produce the correct regular expression upon compiling.
+
+
+[[aws-serverless-multiline-line-continuations]]
+=== Line continuations
+
+Several programming languages use the backslash (`\`) character at the end of a line to denote that the line continues, as in this example:
+
+[source, c]
+----
+printf ("%10.10ld  \t %10.10ld \t %s\
+  %f", w, x, y, z );
+----
+
+To consolidate these lines into a single event, use the following multiline configuration:
+
+[source, yaml]
+----
+multiline:
+  type: pattern
+  pattern: '\\\\$'
+  negate: false
+  match: after
+----
+
+This configuration merges any line that ends with the `\` character with the line that follows it.
+
+NOTE: Note that you should escape the opening square bracket (`[`) in the regular expression, because it specifies a character class i.e. a set of characters that you wish to match. You also have to escape the backslash (`\`) used for escaping the opening square bracket as raw strings are not used. Thus, `^\\[` will produce the required regular expression upon compiling.
```

Comment:
the `NOTE` comes from the different section
the original one is
```
*Notes*: beware that we should escape the backslash we want to match a string literals (`\`) twice, since we are not using raw string. The final result `\\\\$` will produce the regular expression to be compiled properly.
```

Reply:
Good spot, thanks - was trying to make these notes consistent. I've fixed in new transform- topic that now contains the multiline content.